src/main/kotlin/ResponseBuilder.kt

```diff
@@ -1,16 +1,17 @@
 package response
 
-class ResponseBuilder(private val statusCode: Int, private val body: String = "") {
+class ResponseBuilder(private val statusCode: Int,
+                      private val body: String = "",
+                      private val headers: Map<String, List<String>> = mapOf()
+                      ) {
     private val protocol = "HTTP/1.1"
     private val crlf = "\r\n"
 
     fun build(): String {
         val responseLine = generateResponseLine()
-        if (body.isNotEmpty()) {
-            val contentLength = setContentLength()
-            return responseLine + "$contentLength$crlf$crlf$body"
-        }
-        return responseLine
+        var responseHeaders = if (headers.isNotEmpty()) generateHeaders() else ""
```

Comment:
This conditional is unnecessary, since `generateHeaders()` will return an empty string if `headers` is empty.

Reply:
Thank you for pointing this out! I've removed the conditional and just have the function call